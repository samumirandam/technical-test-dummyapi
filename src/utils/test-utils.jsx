import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import reduxThunk from "redux-thunk";
import { configureStore as configureReduxStore } from "@reduxjs/toolkit";
import { BrowserRouter } from "react-router-dom";

import reducer from "@reducers";

const middlewares = [reduxThunk];
const mockStore = configureStore(middlewares);

const AllTheProviders = ({ children }) => {
  const store = configureReduxStore({
    reducer: reducer,
  });
  const setupStore = { ...store.getState(), ...children.props };
  const compomentStore = mockStore(setupStore);
  return (
    <Provider store={compomentStore}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
