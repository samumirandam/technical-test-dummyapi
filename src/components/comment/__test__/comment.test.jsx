import React from "react";
import { render, screen } from "@utils/test-utils";

import Comment from "../index";

const defaultProps = {
  owner: {
    id: "60d0fe4f5311236168a109df",
    title: "mrs",
    firstName: "Anaelle",
    lastName: "Dumas",
    picture: "https://randomuser.me/api/portraits/med/women/25.jpg",
  },
  message: "comment test",
  publishDate: "2022-08-25T23:48:20.035Z",
};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<Comment {...setupStore} />, {});
};

describe("Test for Comment component", () => {
  test("Should render without errors", () => {
    setup();
    expect(screen.getByText("comment test")).toBeTruthy();
  });

  test("Should renders the same component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
