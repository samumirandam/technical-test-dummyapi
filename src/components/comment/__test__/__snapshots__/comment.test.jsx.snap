// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Comment component Should renders the same component 1`] = `
<div>
  <div
    class="Comment"
  >
    <div
      class="UserImage"
    >
      <figure>
        <img
          alt="Image Anaelle Dumas"
          src="https://randomuser.me/api/portraits/med/women/25.jpg"
        />
      </figure>
      <p>
        Anaelle Dumas
      </p>
    </div>
    <p
      class="Comment__date"
    >
      about 6 hours
    </p>
    <p
      class="Comment__message"
    >
      comment test
    </p>
  </div>
</div>
`;
